lib/eslint.js

```diff
@@ -407,6 +407,27 @@ function isDisabledByReportingConfig(reportingConfig, ruleId, location) {
 }
 
 /**
+ * Normalize ECMAScript version from the initial config
+ * @param  {number} ecmaVersion ECMAScript version from the initial config
+ * @param  {boolean} isModule Whether the source type is module or not
+ * @returns {number} normalized ECMAScript version
+ */
+function normalizeEcmaVersion(ecmaVersion, isModule) {
+
+    // need at least ES6 for modules
+    if (isModule && (!ecmaVersion || ecmaVersion < 6)) {
+        ecmaVersion = 6;
+    }
+
+    // calculate ECMAScript edition number from official year version
```

Comment:
I would love a comment explaining why 2009 specifically. Even just something like "(ES6 = ES2015, so subtract 2009)" would help a lot, in my view. But I don't feel that strongly about it.